cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -220,11 +219,19 @@ public ProcessResult run(ProcessMode processMode) {
    * @param is {@link InputStream}.
    * @return {@link CompletableFuture}.
    */
-  private static CompletableFuture<List<String>> readInputStream(InputStream is) {
+  private static CompletableFuture<List<String>> readInputStream(InputStream is, boolean errorStream, List<OutputMessage> outputMessages) {
 
     return CompletableFuture.supplyAsync(() -> {
 
       try (InputStreamReader isr = new InputStreamReader(is); BufferedReader br = new BufferedReader(isr)) {
+
+        String line;
+        while ((line = br.readLine()) != null) {
+          synchronized (outputMessages) {
```

Comment:
This may create a DeadLock.
Consider using a concurrent collection.
I would suggest to use `ConcurrentLinkedQueue` for collecting the `OutputMessage`s and when completed, convert to a regular `List`.

Reply:
done

Reply:
OMG. Sorry, if I caused a mess here.
Lets have a look at this in January after vacation time.
Maybe we can resolve the problem and make it work deterministic and reliable...
In the worst case, we have to revert the Concurrent-Collection approach and need to go back to your original solution using `synchronized` (or better using `ReentrantLock`).